cfgov/ask_cfpb/scripts/export_ask_data.py

```diff
@@ -0,0 +1,143 @@
+
+
+def clean_and_strip_spanish(data):
+    unescaped = html_parser.unescape(data)
+    return html.strip_tags(
+        unescaped).strip().replace(
+        '\t', ' ').replace(
+        '\n', ' ').replace(
+        '\r', ' ')
+
+
+def assemble_output(spanish_only=False):
+    answers = Answer.objects.all()
+    output_rows = []
+    if spanish_only:
+        for answer in answers:
+            output = {
+                'ASK_ID'.encode('UTF16'):
+                str(answer.id).encode('UTF16'),
+                'SpanishURL'.encode('UTF16'): (
+                    answer.spanish_page.url_path.replace(
+                        '/cfgov', '').encode('UTF16')
+                    if answer.spanish_page else ''.encode('UTF16')),
+                'SpanishAnswer'.encode('UTF16'): clean_and_strip_spanish(
+                    answer.answer_es).encode('UTF16')
+            }
+            output_rows.append(output)
+        return output_rows
+    for answer in answers:
+        output = {heading: '' for heading in HEADINGS}
+        output['ASK_ID'] = answer.id
+        output['Question'] = answer.question
+        output['ShortAnswer'] = clean_and_strip(
+            answer.snippet)
+        output['Answer'] = clean_and_strip(
+            answer.answer)
+        # output['SpanishAnswer'] = clean_and_strip(
+        #     answer.answer_es)
+        output['URL'] = answer.english_page.url_path.replace(
+            '/cfgov', '') if answer.english_page else ''
+        output['SpanishURL'] = (
+            answer.spanish_page.url_path.replace(
+                '/cfgov', '') if answer.spanish_page else '')
+        output['Topic'] = answer.category.first().name
+        output['SubCategories'] = " | ".join(
+            [subcat.name for subcat in answer.subcategory.all()])
+        output['Audiences'] = " | ".join(
+            aud.name for aud in answer.audiences.all())
+        output['RelatedQuestions'] = " | ".join(
+            ['{}'.format(a.id) for a in answer.related_questions.all()])
+        output['RelatedResources'] = (
+            answer.next_step.title
+            if answer.next_step
+            else '')
+        output_rows.append(output)
+    return output_rows
+
+
+def export_questions(spanish_only=False):
+    """
+    A rather ridiculous script for exporting Ask CFPB Answer content.
+
+    Run from within cfgov-refresh with:
+    `python cfgov/manage.py runscript export_ask_data`
+
+    CEE staffers use a version of Excel that can't easily import UTF-8
+    non-ascii encodings. Generally the only Ask content that has characters
+    outside the ascii range is Spanish asnwers, so we export the bulk of the
+    data as UTF-8, and Spanish answers as a separate UTF-16 file that our
+    versions of Excel will open with proper diacritical marks.
+    UTF-16 doubles the file size, which can causes Excel performance issues
+    for the full data set. UTF-16 also plays hob with delimiting,
+    pushing text to next cells and making a mess.
```

Comment:
And now I understand the UTF16 encoding above and the need for the spanish break-out in `assemble_output()`! A++ docstringing.